### PR TITLE
Adding sample policy allowed-event-hub-firewall-ip

### PR DESCRIPTION
**Pull request by manishshar**
Adding sample policy for handling event-hub firewall by IPs array.

### CODE REVIEW

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "allowed-event-hub-firewall",
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: please remove "type" and "name"

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "properties": {
+    "displayName": "Allow IP for event hub firewall",
+    "policyType": "Custom",
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: >"policyType": "Custom", [](start = 4, length = 23)

this isn't needed

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "allowed-event-hub-firewall",
+  "properties": {
+    "displayName": "Allow IP for event hub firewall",
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: >Allow IP for event hub firewall [](start = 20, length = 31)

Can you update this to "Event Hub firewall should only allow certain IPs" to match naming standards?

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "policyType": "Custom", 
+    "description": "This policy allows only restricted IPs for Event hub firewall",
+    "mode":"All",
+  "parameters": {
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: >" [](start = 2, length = 1)

please reformat all of these JSON files so the indentation is correct

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      "defaultValue": []
+    }
+  },
+  "policyRule": {
+    "if": {
+      "allOf": [
+        {
+          "field": "type",
+          "equals": "Microsoft.EventHub/namespaces/networkRuleSets"
+        },
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: what if the defaultAction is set to "Allow"?  Then every IP is allowed.  Also, don't you need to take the ipRules.action into account as well?

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        {
+          "not": {
+            "field": "Microsoft.EventHub/namespaces/networkRuleSets/default.ipRules[*].ipMask",
+            "in": "[parameters('AllowedIPs')]"
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: >A [](start = 32, length = 1)

'allowedIps' for consistency

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Allowed only selected IPs for Event hub firewall
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >ed [](start = 7, length = 2)

Allow

Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >Event hub [](start = 32, length = 9)

Event Hub

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+This policy allows user to specify selected IPs for event hub firewall from specified array of IPs. You specify an array of predefined IPs for event hub firewall.
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >This policy allows user t [](start = 0, length = 25)

please make this more clear that the policy only restricts what IPs can be added to the firewall, it doesn't automatically add them to the firewall (nor does it require ALL of them to be there)

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+$definition
+$assignment = New-AzPolicyAssignment -Name <assignmentname> -Scope <scope> -PolicyDefinition $definition
+$assignment 
+````
+
+## Try with CLI
+
+````cli
+
+az policy definition create --name 'allowed-event-hub-firewall' --display-name 'Allow IP for event hub firewall' --description 'List of IPs allowed for event hub firewall' --rules 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.rules.json' --params 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.parameters.json' --mode Indexed
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >Indexed [](start = 465, length = 7)

All

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+## Try on Portal
+
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#blade/Microsoft_Azure_Policy/CreatePolicyDefinitionBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-policy%2Fmaster%2Fsamples%2FEventHub%2Fallowed-event-hub-firewall-ip%2Fazurepolicy.json)
+
+## Try with PowerShell
+
+````powershell
+$definition = New-AzPolicyDefinition -Name 'allowed-event-hub-firewall' -DisplayName 'Allow IP for event hub firewall' -description 'List of IPs allowed for event hub firewall' -Policy 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.rules.json' -Parameter 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.parameters.json' -Mode All
+$definition
+$assignment = New-AzPolicyAssignment -Name <assignmentname> -Scope <scope> -PolicyDefinition $definition
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >n [](start = 103, length = 1)

The example should pass the allowedIps parameter (same for CLI)

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+$assignment 
+````
+
+## Try with CLI
+
+````cli
+
+az policy definition create --name 'allowed-event-hub-firewall' --display-name 'Allow IP for event hub firewall' --description 'List of IPs allowed for event hub firewall' --rules 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.rules.json' --params 'https://raw.githubusercontent.com/Azure/azure-policy/master/samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.parameters.json' --mode Indexed
+
+az policy assignment create --name <assignmentname> --scope <scope> --policy "'allowed-event-hub-firewall"
```
Comment (pilor) [inline samples/EventHub/allowed-event-hub-firewall-ip/README.md]: >' [](start = 78, length = 1)

extra quote here

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 1 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 2 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 3 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 4 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 5 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Added for below for considering only Deny conditions in policyrule
{
"field": "Microsoft.EventHub/namespaces/networkRuleSets/default.defaultAction",
"equals": "Deny
 }

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 6 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md (hunk 7 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: Done

Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md (hunk 9 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md (hunk 10 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md (hunk 11 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: Done

File: samples/EventHub/allowed-event-hub-firewall-ip/README.md (hunk 8 above)
Comment (manishshar) [reply samples/EventHub/allowed-event-hub-firewall-ip/README.md]: changed to - This policy will restrict adding firewall IPs in event hub and allow only those IPs which are predefined in array of selected IPs.

File: samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json (hunk 5 above)
Comment (pilor) [reply samples/EventHub/allowed-event-hub-firewall-ip/azurepolicy.json]: Thats not really what you want though right?  As written this policy will ONLY deny if defaultAction=deny and ipRules contains an IP not in allowedIPs.  The title and the description imply that if this policy was assigned ONLY a subset (or all) of the allowedIPs should have access to an event hub.  The policy should deny if defaultAction=allow

---
In reply to: [310361757](https://github.com/Azure/azure-policy/pull/448#discussion_r310361757) [](ancestors = 310361757)

**Pull request merged**